### ISSUE TITLE
QA Note: For Testing Changes to Scoring + Firebase

### ISSUE THREAD

**Issue by helencarey** [ByunLab/staRt-ionic#487]
**Changes affect:** All Quest instances. Highscores are only saved for Quest sessions.

**Tests:** These tests ensure that highscores data is properly being saved to firebase in several contexts. This tickets tests work in #423.

**Notes:** We will release branch internally only, so that the debug box can be displayed.

-----------------------
**Please do:** 

**Step 1: Short off-protocol session** 
- Go to quest and select 20 trials (any /r/ variants)
- Complete the Quest and generate some new highscores.  
- Go to Profiles and switch away from the user, and then back to the user who just completed the Quest. 
- Start a new Quest
- [x] Are the highscores from the previous session being read-in correctly?

**Step 2: Complete an on-protocol session** 
- Start an on-protocol session. After you are taken to Free Play, click into Quest -- no need to wait 5 min.
- Select Word on the Quest menu-page. On the Quest config-page, select any /r/ variants and 100 trials should automatically be selected. Hit start.
- Generate some highscores and complete the quest.
- Switch active profiles, and then switch back user who just completed the Quest. 
- Start the next on-protocol session. 
- After you are taken to Free Play, click into Quest -- no need to wait 5 min.
- Select Word on the Quest menu-page. On the Quest config-page, select any /r/ variants and 100 trials should automatically be selected. 
- Hit start. Don’t navigate away. You’ll continue in the session for Step 3.
- [x] Take a screenshot; you’ll need it for Step 4.
- [x] Are the highscores from the previous session being read-in correctly?

**Step 3: Resume an on-protocol session  (continued from above)**
- Complete 50 trials of an on-protocol Quest session. Make note of the scores in the debug box, and then hit Stop Quest. 
- Go back to Profiles. Switch active profiles, and then switch back and resume the on-protocol session.
- [x] Are the scores listed correctly in the debug box? Especially:
  - [x] Most Gold in Block
  - [x] High Score in Block
  - [x] Word Difficulty
  - [x] Word Performance
- [x] Do the coins return? 
- [x] No dialog boxes should come up when you resume the session. Is this what happened? 

**Step 4: Switch between on-protocol and off-protocol quests (continued from above)**
- Now end the on-protocol session from step 3. 
- Go back to Profiles. (The purple button should read ‘Start Session.’)
- Start an off-protocol session, by selecting Quest directly -- do not hit the purple btn.
  - Menu Page: select ‘Word.’ 
  - Config Page: select any /r/ variants and any # of trial. 
  - Hit Begin Quest.
- Look at the Score debug box.
  - [x] Do the highscores match those from Step 2? 
  (The Debug Box should reflect highscores from the last completed Quest, and not highscores achieved during saved-but-uncompleted Quests.)

**Step 5. Existing user with cleared scores (continued from above)**
- Hit ‘Reset Highscores.’ The quest should end automatically.
- Start a new off-protocol Quest with 20 trials. 
- [ ] The scores should all be 0.
- Complete the Quest, and return to Profiles.
- Switch active profiles, and then switch back user who just completed the Quest. 
- Start a new off-protocol Quest. 
- [x] Are the highscores from the previous session being read-in correctly?

**Step 6. Resuming a session after clearing cleared scores (continued from above)**
- Complete 1 block (10 trials) and then hit ‘Reset Highscores.’
- Return to Profiles. Switch active profiles, and then switch back user from Step 5.
- Go to Profiles/Recording and resume the last session.
- [x] The scores should only reflect the coins on-screen. 
  In other words, if you had 3 golds in the block, your Most Gold in Block (mgib) and Most Gold in Quest (mgiq) should both have highscores of 3.

**Step 7. Saving with New Account** 
- Create a new user.
- Do an off-protocol quest with 20 trials, and generate some new highscores.
- Return to Profiles. Switch active profiles, and then switch back user who just completed the Quest. 
- Start a new off-protocol quest. 
- [x] Are the highscores from the previous session being read-in correctly?

**Comment by yl5504** [comment]
@helencarey I am wondering - shall I do the on-protocol test under "speedy" or any name is ok? Cuz if I start an on-protocol session under a randomly-named participant, it will navigate me to the Quiz. if I start an on-protocol session under "Speedy", it will navigate me to Free Play then Quest properly.
![IMG_0032](https://user-images.githubusercontent.com/43120402/70666761-57cf2f00-1c3d-11ea-9402-de9ba044941b.PNG)
![IMG_0033](https://user-images.githubusercontent.com/43120402/70666762-57cf2f00-1c3d-11ea-8502-fa39b85cbf24.PNG)

**Comment by helencarey** [comment]
@yl5504 Yup, Speedy is fine, as long as it lets you adjust the number of trials. 
--------
The other named users should work too, however you will have to do the quizzes 1 time before they take you to the Quest Sessions. (When you are on-protocol, the quizzes are like pre-assessment tests. After you've done them once, then you unlock the 'First Session.')

**Comment by yl5504** [comment]
![IMG_0036](https://user-images.githubusercontent.com/43120402/70746732-675d7f00-1cf4-11ea-887a-e460af2f289f.PNG)
Step 5. After I reset highscore and then start a new off-protocol quest, the scores look like this - not every one is 0. Is that a problem? I am a bit confused. @helencarey

**Comment by helencarey** [comment]
@yl5504 Good catch. Actually, what you are seeing is correct. Those are the minimum values for those categories (hsib & hsiq). I forgot to put that in my note.
